server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -312,6 +318,9 @@ public void onFailure(Exception e) {
                 OriginalIndices indices = entry.getValue();
                 SearchRequest ccsSearchRequest = SearchRequest.subSearchRequest(searchRequest, indices.indices(),
                     clusterAlias, timeProvider.getAbsoluteStartMillis(), false);
+                // NORELEASE: We should only set the parent task if the target node on the new version;
+                // otherwise, this sub task will be cancelled when some nodes get removed in the remote cluster.
+                ccsSearchRequest.setParentTask(parentTaskId);
```

Comment:
I assume that nothing needs to be done for the case where minimizing roundtrips is disabled as we send out shard requests that already have their parent task id set?

Reply:
Good question. I think we also need to trip the parent task from those shard requests if the remote cluster is on an old version (i.e., it cancels child tasks if nodes in its cluster leave). I think we should fix it after having a task heartbeat.